retirement/serializers.py

```diff
@@ -558,32 +566,44 @@ def update(self, instance, validated_data):
                     # coupon on the new orderline created. In other words, any
                     # coupon used for the initial purchase is applied again
                     # here.
-                    need_transaction = True
-                    amount = (
-                        validated_data['retreat'].price -
-                        order_line.coupon_real_value
-                    )
-                    if not (payment_token or single_use_token):
-                        raise serializers.ValidationError({
-                            'non_field_errors': [_(
-                                "The new retreat is more expensive than "
-                                "the current one. Provide a payment_token or "
-                                "single_use_token to charge the balance."
-                            )]
-                        })
+                    # need_transaction = True
+                    # amount = (
+                    #     validated_data['retreat'].price -
+                    #     order_line.coupon_real_value
+                    # )
+                    # if not (payment_token or single_use_token):
+                    #     raise serializers.ValidationError({
+                    #         'non_field_errors': [_(
+                    #             "The new retreat is more expensive than "
+                    #             "the current one. Provide a payment_token or"
+                    #             " single_use_token to charge the balance."
+                    #         )]
+                    #     })
+                    raise serializers.ValidationError({
+                        'non_field_errors': [_(
+                            "You can only exchange for a retreat with the "
+                            "same price."
+                        )]
+                    })
                 if current_retreat.price > new_retreat_price:
                     # If a coupon was applied for the purchase, check if the
                     # real cost of the purchase was lower than the price
                     # difference.
                     # If so, refund the real cost of the purchase.
                     # Else refund the difference between the 2 retreats.
-                    need_refund = True
-                    price_diff = (
-                        current_retreat.price -
-                        validated_data['retreat'].price
-                    )
-                    real_cost = order_line.total_cost
-                    amount = min(price_diff, real_cost)
+                    # need_refund = True
+                    # price_diff = (
+                    #     current_retreat.price -
+                    #     validated_data['retreat'].price
```

Comment:
No need to keep commented code without any explanation

Reply:
code removed and linked to git commit